packages/server/src/automations/steps/updateRow.ts

```diff
@@ -84,9 +84,12 @@ export async function run({ inputs, appId, emitter }: AutomationStepInput) {
 
   // clear any undefined, null or empty string properties so that they aren't updated
   for (let propKey of Object.keys(inputs.row)) {
+    const clearRelationships =
+      inputs.meta?.fields?.[propKey]?.clearRelationships
     if (
-      (inputs.row[propKey] == null || inputs.row[propKey] === "") &&
-      !inputs.meta?.fields?.[propKey]?.clearRelationships
+      ((inputs.row[propKey] == null || inputs.row[propKey] === "") &&
```

Comment:
This is quite confusing. Why we have different logic for `undefined` and `false` `clearRelationships`s?

Reply:
I think I just got a bit confused - I've refactored to simplify it